libcaf_net/test/ip.cpp

```diff
@@ -30,16 +30,48 @@
 using namespace caf;
 using namespace caf::net;
 
-CAF_TEST_FIXTURE_SCOPE(ip_tests, host_fixture)
+namespace {
+
+struct fixture : host_fixture {
+  fixture() : v6_local{{0}, {0x1}} {
+    v4_local = ip_address{make_ipv4_address(127, 0, 0, 1)};
+    v4_any_addr = ip_address{make_ipv4_address(0, 0, 0, 0)};
+  }
+
+  bool contains(ip_address x) {
+    return std::count(addrs.begin(), addrs.end(), x) > 0;
+  }
+
+  ip_address v4_any_addr;
+  ip_address v6_any_addr;
+  ip_address v4_local;
+  ip_address v6_local;
+
+  std::vector<ip_address> addrs;
+};
+
+} // namespace
+
+CAF_TEST_FIXTURE_SCOPE(ip_tests, fixture)
 
 CAF_TEST(resolve) {
-  ip_address v4_local{make_ipv4_address(127, 0, 0, 1)};
-  ip_address v6_local{{0}, {0x1}};
-  auto addrs = ip::resolve("localhost");
+  addrs = ip::resolve("localhost");
+  CAF_CHECK(!addrs.empty());
+  CAF_CHECK(contains(v4_local) || contains(v6_local));
```

Comment:
Doesn't this fail on systems without IPv6 enabled?

Reply:
I hope the `||` will prevent that. It will fail if there is neither v4 nor v6.